contentcuration/contentcuration/frontend/channelEdit/views/NodePanel.vue

```diff
@@ -45,13 +48,16 @@
 <script>
 
   import { mapActions, mapGetters } from 'vuex';
+
+  import { RouterNames } from '../constants';
+  import ContentNodeEditListItem from '../components/ContentNodeEditListItem/ContentNodeEditListItem';
```

Comment:
Isn't the convention to name the file `index.vue` if setting it in it's own directory? And thereby, the reasoning for why the parent directory is named like a component. Also, I believe the reasoning to put it into it's own directory is to also group other dependent templates within that, but these appear to be standalone. So an alternate structure would look like this:
```
components/contentNodeList/ContentNodeListItem.vue
components/contentNodeList/ContentNodeEditListItem.vue
```

Reply:
> Isn't the convention to name the file index.vue

That sounds neat, though I am not sure about how to treat spec files that usually have the same name then? Any experience/suggestions? The only thing I'd suggest is to keep all related files in one directory and other than that I don't have a strong opinion.

 > components/contentNodeList/ContentNodeListItem.vue
components/contentNodeList/ContentNodeEditListItem.vue

I don't have an opinion on this at all. The only reason I am keeping things in plain structure is that we don't have `TreeView` in `pages` yet and soon we'll be hopefully moving everything to `pages`/`components` and restructuring it all. When we discussed this with @devon and others I think that there was a recommendation to place only components shared within particular SPA pages to `components` directory and to place components that are used only in a specific page to that page directory. In this case it would probably looked like


```
channelEdit/components/ContentNodeListItem.vue (used in both TreeView and StagingTreeView)

channelEdit/pages/TreeView/index.vue
channelEdit/pages/TreeView/ContentNodeEditListItem.vue (used only in TreeView)
```


I guess? If I am not mistaken the reasoning behind this was to keep track of various dependencies. @indirectlylit any thoughts? I am not 100% sure about these details so it would be clear on it in general before we move it all. I can solve it here or it can be done here #1924.

Reply:
>  @bjester: Isn't the convention to name the file `index.vue` if setting it in its own directory?

Yes, the advantage is that you can start initially with `SomeComponent.vue`, and then later switch to `SomeComponent/index.vue` as things get more complicated, and your external import statements don't need to change.

> @MisRob: I am not sure about how to treat spec files that usually have the same name then?

I think either `SomeComponent/index.spec.vue` or `SomeComponent/SomeComponent.spec.vue` would be fine?

>  @bjester: I believe the reasoning to put it into it's own directory is to also group other dependent templates within that, but these appear to be standalone

>  @MisRob: I think that there was a recommendation to place only components shared within particular SPA pages to components directory and to place components that are used only in a specific page to that page directory
> * `channelEdit/components/ContentNodeListItem.vue` (used in both `TreeView` and `StagingTreeView`)
> * `channelEdit/pages/TreeView/index.vue`
> * `channelEdit/pages/TreeView/ContentNodeEditListItem.vue` (used only in `TreeView`)

Yes, I think you're both correct and maybe saying similar things thing here?

What you've outlined covers all the key conventions. The only edge I've come across in Kolibri are situations where you have clusters of "related" components that are often used together, but don't necessarily fit the parent/child relationship.

For example, one could rely on having a common `Tab*` prefix for grouping:

* `components/TabGroup.vue`
* `components/TabButton.vue`

Or, one could add another level of organization:

* `components/tabs/TabGroup.vue`
* `components/tabs/TabButton.vue`

I can see arguments for both strategies. We used to use the sub-directory strategy in Kolibri, but in the design system I started migrating to the common-prefix strategy to keep things flat and easier to scan.

Reply:
> I can see arguments for both strategies. We used to use the sub-directory strategy in Kolibri, but in the design system I started migrating to the common-prefix strategy to keep things flat and easier to scan.

Yeah I am not sure about sub-directories either but slightly tend to what @bjester suggests (while being careful to not end up with 5 levels of nested components) as for a general design system I assume that prefixes might be much shorter (`Btn-`, `Form-`, `Tab-`) when compared to specific use-cases within an app (`ContentNodeList-`)

Reply:
So for now I renamed files in a component dir to _index.js_ and _index.spec.js_ and leaving it plain because there will be only one of those components left in _/components_ (_ContentNodeEditListItem_ will be moved to _pages/TreeView_ as soon as we have the page there). I am also linking this conversation to #1924.